packages/jazz-tools/src/react-core/hooks.ts

```diff
@@ -634,53 +656,413 @@ export function useAccount<
+    setIsLoaded(value.$isLoaded);
+  }, []);
+
+  const ref = useSubscriptionRef(subscription, {
+    onUpdate,
+  });
+
+  return [
+    useSubscriptionSelector<S, R, TSelectorReturn>(subscription, options),
+    Object.defineProperty(ref, "$isLoaded", {
+      value: isLoaded,
+    }) as MaybeLoadedCoValueRef<Loaded<S, R>>,
+  ];
+}
+
+/**
+ * Returns both a reactive state and a ref for the current account.
+ *
+ * This hook combines `useAccount` and `useAccountRef` to provide:
+ * - **State**: Reactive value that triggers re-renders only when selected fields change
+ * - **Ref**: Non-reactive reference for accessing the full account in callbacks/event handlers
+ *
+ * This is useful when you need to:
+ * - Display specific account data in your component (using state with a selector)
+ * - Edit other account properties or use the $jazz API in event handlers without causing re-renders
+ *
+ * @example
+ * ```tsx
+ * function AccountSettings() {
+ *   // Only re-render when profile.name changes
+ *   const [profileName, accountRef] = useAccountAndRef(Account, {
+ *     resolve: { profile: true },
+ *     select: (account) => account.$isLoaded ? account.profile.name : undefined,
+ *   });
+ *
+ *   if (profileName === undefined) {
+ *     return <div>Loading account...</div>;
+ *   }
+ *
+ *   return (
+ *     <div>
+ *       <h1>Welcome, {profileName}</h1>
+ *       <button
+ *         onClick={() => {
+ *           const currentAccount = accountRef.current;
+ *           if (currentAccount.$isLoaded) {
+ *             const currentStatus = currentAccount.profile.status;
+ *             currentAccount.profile.$jazz.set("status", currentStatus === "active" ? "inactive" : "active");
+ *           }
+ *         }}
+ *       >
+ *         Toggle Status
+ *       </button>
+ *     </div>
+ *   );
+ * }
+ * ```
+ */
+export function useAccountAndRef<
+  A extends AccountClass<Account> | AnyAccountSchema,
+  // @ts-expect-error we can't statically enforce the schema's resolve query is a valid resolve query, but in practice it is
+  const R extends ResolveQuery<A> = SchemaResolveQuery<A>,
+  TSelectorReturn = MaybeLoaded<Loaded<A, R>>,
+>(
+  /** The account schema to use. Defaults to the base Account schema */
+  AccountSchema: A = Account as unknown as A,
+  /** Optional configuration for the subscription */
+  options?: UseCoValueOptions<A, R, TSelectorReturn>,
+): [TSelectorReturn, CoValueRef<MaybeLoaded<Loaded<A, R>>>] {
+  const subscription = useAccountSubscription(AccountSchema, options);
+  const [isLoaded, setIsLoaded] = React.useState(
+    () => getSubscriptionValue(subscription).$isLoaded,
+  );
+
+  const onUpdate = useCallback((value: MaybeLoaded<Loaded<A, R>>) => {
+    setIsLoaded(value.$isLoaded);
+  }, []);
+
+  const ref = useSubscriptionRef(subscription, {
+    onUpdate,
+  });
+
+  return [
+    useSubscriptionSelector<A, R, TSelectorReturn>(subscription, options),
+    Object.defineProperty(ref, "$isLoaded", {
```

Comment:
Technically speaking, `$isLoaded` is not guaranteed to be in-sync with the ref's value. In practice, since we update the ref and the `isLoaded` state at the same time, if any change triggers a change in `$isLoaded` to be `false`, then consuming children component would unmount/re-render before they get the chance for anything to go out of sync.

```tsx
  {coValueRef.$isLoaded && (
    <Child
      playlistRef={coValueRef}
    />
  )}
```